src/GWallet.Backend/JsonRpcSharp.fs

```diff
@@ -114,11 +130,26 @@ module JsonRpcSharp =
                 |> Encoding.UTF8.GetBytes
                 |> ArraySegment<byte>
 
-            let! bytesReceived = socket.SendAsync(buffer, SocketFlags.None) |> Async.AwaitTask
+            let! bytesSent = socket.SendAsync(buffer, SocketFlags.None) |> Async.AwaitTask
             let pipe = Pipe()
-            do! FillPipeAsync socket pipe.Writer
-            let! str = ReadPipe pipe.Reader
-            return str
+            let w = FillPipeAsync socket pipe.Writer
+            let r = ReadPipe pipe.Reader
+            // wait for both tasks finished
+            let taskList = [w;r] 
+                           |> Async.Parallel
+                           |> Async.RunSynchronously
```

Comment:
Mmm I'm not sure I like the use of Async.RunSynchronously inside an async{} block. How about, instead of these 3 lines above, use these 2 lines below:

```
let writerAndReaderJobs = Async.Parallel ([w;r])
let! taskList = writerAndReaderJobs
```